### PR TITLE
feat(data-access): get more data from falconer

### PR DESCRIPTION
**Pull request by hadzhehsen**

### CODE REVIEW

File: apps/haqq-website/app/(main)/blog/page.tsx [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import type { Metadata } from 'next';
-import { getBlogPosts } from '../../../utils/get-blog-posts';
+import { getHAQQBlockPostsFromFalconer } from '../../../utils/get-blog-posts';
```
Comment (olegshilov) [inline apps/haqq-website/app/(main)/blog/page.tsx]: blog, not block

File: apps/haqq-website/app/(main)/blog/[slug]/page.tsx [hunk 2]
```diff
@@ -9,7 +12,7 @@ export const dynamicParams = true;
 export const revalidate = REVALIDATE_TIME;
 
 export async function generateStaticParams() {
-  const { posts } = await getBlogPosts();
+  const { posts } = await getHAQQBlockPostsFromFalconer();
```
Comment (olegshilov) [inline apps/haqq-website/app/(main)/blog/[slug]/page.tsx]:
```suggestion
  const { posts } = await getHaqqBlogPostsFromFalconer();
```

File: libs/data-access/falconer/src/lib/haqq/get-blog-posts.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import { FALCONER_ENDPOINT } from '../../constants';
+import { FalconerRequestInit } from '../../types';
+
+interface Post {
```
Comment (olegshilov) [inline libs/data-access/falconer/src/lib/haqq/get-blog-posts.ts]:
```suggestion
export interface BlogPost {
```

File: libs/data-access/falconer/src/lib/haqq/get-blog-posts.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  slug: string;
+  date: string;
+  content?: string;
+  description?: string;
+  image: { src: string; width: number; height: number } | null;
+  isFeatured?: boolean;
+  tags: string[];
+}
+
+export async function getHAQQBlogPosts(options: Partial<FalconerRequestInit>) {
```
Comment (olegshilov) [inline libs/data-access/falconer/src/lib/haqq/get-blog-posts.ts]:
```suggestion
export async function getHaqqBlogPosts(options: Partial<FalconerRequestInit>) {
```

File: libs/data-access/falconer/src/lib/haqq/get-partners.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
+import { FALCONER_ENDPOINT } from '../../constants';
+import { FalconerRequestInit } from '../../types';
+
+interface Partner {
```
Comment (olegshilov) [inline libs/data-access/falconer/src/lib/haqq/get-partners.ts]:
```suggestion
export interface Partner {
```

File: libs/data-access/falconer/src/lib/haqq/get-partners.ts [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+  SERVICE = 'service',
+  INSTITUTIONAL = 'institutional',
+}
+
+export enum PartnerStatus {
+  LIVE = 'live',
+  PLANNED = 'planned',
+}
+
+export async function getHAQQPartners(options: Partial<FalconerRequestInit>) {
```
Comment (olegshilov) [inline libs/data-access/falconer/src/lib/haqq/get-partners.ts]:
```suggestion
export async function getHaqqPartners(options: Partial<FalconerRequestInit>) {
```

File: libs/data-access/falconer/src/lib/haqq/get-partners.ts [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    method: 'GET',
+    ...options,
+  });
+
+  if (!response.ok) {
+    throw new Error('Partners fetch failed');
+  }
+
+  const responseJson = await response.json();
+  console.log({ responseJson });
```
Comment (olegshilov) [inline libs/data-access/falconer/src/lib/haqq/get-partners.ts]: cleanup and remove unnecessary log output

File: libs/data-access/falconer/src/lib/islamic/get-disclaimer.ts [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+import { FalconerRequestInit } from '../../types';
+
+export async function getIslamicDisclaimer(
+  options: Partial<FalconerRequestInit>,
+  locale: string,
+) {
+  const requestUrl = new URL(
+    `/islamic/disclaimer?locale=${locale}`,
+    FALCONER_ENDPOINT,
+  );
```
Comment (olegshilov) [inline libs/data-access/falconer/src/lib/islamic/get-disclaimer.ts]:
```suggestion
  const requestUrl = new URL(
    '/islamic/disclaimer',
    FALCONER_ENDPOINT,
  );
  requestUrl.searchParams.append('locale', locale);
```

Comment (olegshilov) [reply libs/data-access/falconer/src/lib/islamic/get-disclaimer.ts]: here and everywhere in this pr where you request localised data

File: libs/data-access/falconer/src/lib/islamic/get-roadmap.ts [hunk 9]
```diff
@@ -0,0 +1,30 @@
+import { FALCONER_ENDPOINT } from '../../constants';
+import { FalconerRequestInit } from '../../types';
+
+interface RoadmapPeriod {
```
Comment (olegshilov) [inline libs/data-access/falconer/src/lib/islamic/get-roadmap.ts]:
```suggestion
export interface RoadmapPeriod {
```

File: libs/islamic-website/roadmap-page/src/lib/roadmap-page/roadmap-page.tsx [hunk 10]
```diff
@@ -179,7 +179,7 @@ export function RoadmapPage({
           </h1>
 
           <div className="mt-[48px] flex flex-col gap-y-[12px] md:mt-[44px] lg:mt-[60px] lg:gap-y-[20px]">
-            {roadmap.map((period, idx) => {
+            {roadmap?.map((period, idx) => {
```
Comment (olegshilov) [inline libs/islamic-website/roadmap-page/src/lib/roadmap-page/roadmap-page.tsx]: why?

**Review by olegshilov — CHANGES_REQUESTED** [review]

File: apps/haqq-website/app/(main)/ecosystem/page.tsx [hunk 11]
```diff
@@ -1,5 +1,5 @@
 import type { Metadata } from 'next';
-import { getPartners } from '../../../utils/get-partners';
+import { getHAQQPartnersFromFalconer } from '../../../utils/get-partners';
```
Comment (olegshilov) [inline apps/haqq-website/app/(main)/ecosystem/page.tsx]:
```suggestion
import { getHaqqPartnersFromFalconer } from '../../../utils/get-partners';
```

File: apps/haqq-website/utils/get-partners.ts [hunk 12]
```diff
@@ -1,67 +1,21 @@
...
-
-export const getPartners = cache(async () => {
-  const { storyblokApi } = storyblokInit({
-    accessToken: STORYBLOK_ACCESS_TOKEN,
-    use: [apiPlugin],
-  });
-
-  if (!storyblokApi) {
-    throw new Error('Failed to init storyblok');
+export const getHAQQPartnersFromFalconer = cache(async () => {
```
Comment (olegshilov) [inline apps/haqq-website/utils/get-partners.ts]:
```suggestion
export const getHaqqPartnersFromFalconer = cache(async () => {
```

File: libs/data-access/falconer/src/lib/haqq/get-blog-posts.ts [hunk 13]
```diff
@@ -0,0 +1,30 @@
+import { FALCONER_ENDPOINT } from '../../constants';
+import { FalconerRequestInit } from '../../types';
+
+export interface Post {
```
Comment (olegshilov) [inline libs/data-access/falconer/src/lib/haqq/get-blog-posts.ts]:
```suggestion
export interface BlogPost {
```

File: libs/data-access/falconer/src/lib/islamic/get-disclaimer.ts [hunk 14]
```diff
@@ -0,0 +1,24 @@
+import { FALCONER_ENDPOINT } from '../../constants';
+import { FalconerRequestInit } from '../../types';
+
+export async function getIslamicDisclaimer(
+  options: Partial<FalconerRequestInit>,
+  locale: string,
+) {
+  const requestUrl = new URL('/islamic/disclaimer/', FALCONER_ENDPOINT);
```
Comment (olegshilov) [inline libs/data-access/falconer/src/lib/islamic/get-disclaimer.ts]:
```suggestion
  const requestUrl = new URL('/islamic/disclaimer', FALCONER_ENDPOINT);
```

**Review by olegshilov — CHANGES_REQUESTED** [review]

File: apps/haqq-website/app/(main)/blog/page.tsx (hunk 1 above)
Comment (hadzhehsen) [reply apps/haqq-website/app/(main)/blog/page.tsx]: Done

File: apps/haqq-website/app/(main)/blog/[slug]/page.tsx (hunk 2 above)
Comment (hadzhehsen) [reply apps/haqq-website/app/(main)/blog/[slug]/page.tsx]: Done

File: libs/data-access/falconer/src/lib/haqq/get-blog-posts.ts (hunk 3 above)
Comment (hadzhehsen) [reply libs/data-access/falconer/src/lib/haqq/get-blog-posts.ts]: Done

File: libs/data-access/falconer/src/lib/haqq/get-blog-posts.ts (hunk 4 above)
Comment (hadzhehsen) [reply libs/data-access/falconer/src/lib/haqq/get-blog-posts.ts]: Done

File: libs/data-access/falconer/src/lib/haqq/get-partners.ts (hunk 5 above)
Comment (hadzhehsen) [reply libs/data-access/falconer/src/lib/haqq/get-partners.ts]: Done

File: libs/data-access/falconer/src/lib/haqq/get-partners.ts (hunk 6 above)
Comment (hadzhehsen) [reply libs/data-access/falconer/src/lib/haqq/get-partners.ts]: Done

File: libs/data-access/falconer/src/lib/haqq/get-partners.ts (hunk 7 above)
Comment (hadzhehsen) [reply libs/data-access/falconer/src/lib/haqq/get-partners.ts]: Done

File: libs/data-access/falconer/src/lib/islamic/get-disclaimer.ts (hunk 8 above)
Comment (hadzhehsen) [reply libs/data-access/falconer/src/lib/islamic/get-disclaimer.ts]: Done

File: libs/data-access/falconer/src/lib/islamic/get-roadmap.ts (hunk 9 above)
Comment (hadzhehsen) [reply libs/data-access/falconer/src/lib/islamic/get-roadmap.ts]: Done

File: libs/islamic-website/roadmap-page/src/lib/roadmap-page/roadmap-page.tsx (hunk 10 above)
Comment (hadzhehsen) [reply libs/islamic-website/roadmap-page/src/lib/roadmap-page/roadmap-page.tsx]: Removed

File: apps/haqq-website/app/(main)/ecosystem/page.tsx (hunk 11 above)
Comment (hadzhehsen) [reply apps/haqq-website/app/(main)/ecosystem/page.tsx]: Done

File: apps/haqq-website/utils/get-partners.ts (hunk 12 above)
Comment (hadzhehsen) [reply apps/haqq-website/utils/get-partners.ts]: Done

File: libs/data-access/falconer/src/lib/haqq/get-blog-posts.ts (hunk 13 above)
Comment (hadzhehsen) [reply libs/data-access/falconer/src/lib/haqq/get-blog-posts.ts]: Done

File: libs/data-access/falconer/src/lib/islamic/get-disclaimer.ts (hunk 14 above)
Comment (hadzhehsen) [reply libs/data-access/falconer/src/lib/islamic/get-disclaimer.ts]: Done

**Review by olegshilov — APPROVED** [review]

**Pull request merged**